test_tools/process_client.py

```diff
@@ -830,3 +900,160 @@ def ensure_workspace(workspace):
+        map_data.create_map_with_internal_layers_file(map_layers, file_path=file_path, native_extent=native_extent,
+                                                      native_crs=crs)
+        file_paths = [file_path]
+
+    for file_path in file_paths:
+        assert os.path.isfile(file_path), file_path
+    files = []
+    with ExitStack() as stack:
+        data = {}
+        if not with_chunks:
+            for file_path in file_paths:
+                assert os.path.isfile(file_path), file_path
+            files = [('file', (os.path.basename(fp), stack.enter_context(open(fp, 'rb')))) for fp in file_paths]
+        else:
+            data['file'] = [os.path.basename(file) for file in file_paths]
+        if access_rights and access_rights.get('read'):
+            data["access_rights.read"] = access_rights['read']
+        if access_rights and access_rights.get('write'):
+            data["access_rights.write"] = access_rights['write']
+        if title:
+            data['title'] = title
+        if description:
+            data['description'] = description
+        if style_file:
+            files.append(('style', (os.path.basename(style_file), stack.enter_context(open(style_file, 'rb')))))
+        if overview_resampling:
+            data['overview_resampling'] = overview_resampling
+        if time_regex:
+            data['time_regex'] = time_regex
+        if time_regex_format:
+            data['time_regex_format'] = time_regex_format
+        if publication_type == LAYER_TYPE and crs:
+            data['crs'] = crs
+        if external_table_uri:
+            data['external_table_uri'] = external_table_uri
+
+        response = requests.patch(r_url,
+                                  files=files,
+                                  headers=headers,
+                                  data=data,
+                                  timeout=HTTP_TIMEOUT,
+                                  )
+    raise_layman_error(response)
+
+    assert response.json()['uuid'] == uuid, f'uuid={uuid}, response.uuid={response.json().get("uuid")}'
+
+    expected_resp_keys = ['name', 'uuid', 'url']
+    if with_chunks:
+        expected_resp_keys.append('files_to_upload')
+    assert all(key in response.json() for key in expected_resp_keys), f'uuid={uuid}, response={response.json()}'
+
+    if with_chunks and not do_not_upload_chunks:
+        upload_file_chunks_by_uuid(publication_type, uuid, file_paths,)
+
+    if not do_not_upload_chunks:
+        wait_for_publication_status_by_uuid(uuid, publication_type,
+                                            check_response_fn=check_response_fn,
+                                            headers=headers,
+                                            raise_if_not_complete=raise_if_not_complete)
+    wfs.clear_cache()
+    wms.clear_cache()
+    if temp_dir:
+        shutil.rmtree(temp_dir)
+    return response.json()
+
+
+patch_map_by_uuid = partial(patch_publication_by_uuid, MAP_TYPE)
```

Comment:
I think `patch_map` is enough, but I will leave it up to you for all three (patch, get, delete) methods..